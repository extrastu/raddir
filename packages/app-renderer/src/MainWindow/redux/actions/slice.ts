import { createSlice } from '@reduxjs/toolkit';

import { omitUnderscorePrefixActions } from 'src/utils/redux';

import type { StoreState } from '../types';

import { _setAppDetails } from './reducers.app';
import { updateClientAreaSize } from './reducers.DisplayConfig';

export function getInitialState(): StoreState {
    return {
        appDetails: null,
        displayConfig: {
            clientAreaSize: null,
        },
    };
}

const slice = createSlice({
    name: 'Main',
    initialState: getInitialState(),
    reducers: {
        _setAppDetails,
        updateClientAreaSize,
    },
});

export const _actions = slice.actions;
export const _reducer = slice.reducer;

export const actions = omitUnderscorePrefixActions(slice.actions);
