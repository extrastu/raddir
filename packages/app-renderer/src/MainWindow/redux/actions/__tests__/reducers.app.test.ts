import type { AppDetails } from '@tecra/app-common';

import { getInitialState, _actions, _reducer } from '../slice';

{
    const name = '_setAppDetails';

    describe(`Test redux reducer \`${name}\``, () => {
        it('should update app details', () => {
            const prevState = getInitialState();

            const appDetails: AppDetails = {
                name: 'tecra',
                version: 'unknown',
                module: {
                    chrome: 'unknown',
                    electron: 'unknown',
                    node: 'unknown',
                    v8: 'unknown',
                },
            };

            expect(_reducer(prevState, _actions[name](appDetails)).appDetails).toMatchSnapshot();
        });
    });
}
