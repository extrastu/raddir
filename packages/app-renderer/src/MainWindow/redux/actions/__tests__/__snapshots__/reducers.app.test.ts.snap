// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test redux reducer \`_setAppDetails\` should update app details 1`] = `
{
  "module": {
    "chrome": "unknown",
    "electron": "unknown",
    "node": "unknown",
    "v8": "unknown",
  },
  "name": "tecra",
  "version": "unknown",
}
`;
